source/frontend/src/telemetry/metrics.ts

```diff
@@ -0,0 +1,82 @@
+import { metrics } from '@opentelemetry/api';
+import {
+  ATTR_HTTP_RESPONSE_STATUS_CODE,
+  METRIC_HTTP_CLIENT_REQUEST_DURATION,
+  METRIC_HTTP_CLIENT_RESPONSE_BODY_SIZE,
+} from '@opentelemetry/semantic-conventions/incubating';
+
+import { TelemetryConfig } from './config';
+import { isBlocked, isBrowserEnvironment, NETWORK_METER } from './utils';
+
+export const registerNetworkMetrics = (config: TelemetryConfig) => {
+  if (!isBrowserEnvironment() || !window.performance) {
+    return;
+  }
+
+  if (config.debug) {
+    console.info('[INFO] Registering network request metrics');
+  }
+
+  const meter = metrics.getMeter(NETWORK_METER);
+
+  // measure the time it takes to make an XHR request
+  const timeSpentMetric = meter.createHistogram(METRIC_HTTP_CLIENT_REQUEST_DURATION, {
+    description: 'The duration of an outgoing HTTP request.',
+    unit: 's',
+    advice: {
+      explicitBucketBoundaries: [
+        0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10,
```

Comment:
would a comment help for this? also, should this be externally configurable?

Reply:
we could make this configurable - sure. This is lifted from the open-telemetry documentation as the suggested buckets for the histogram distribution. These are the buckets that count how many requests fall in those boundaries (the above is in seconds). We can move that to the tenant.json where we already have some other config

FYI: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation

![image](https://github.com/user-attachments/assets/d6714221-fd9a-40cf-a362-4b0804d48ee4)